packages/web/src/js/ScrollView.ts

```diff
@@ -0,0 +1,127 @@
+// The `ScrollView` component is currently untested. This is because it is not possible to test it in a headless
+// environment (tests fail on `getBoundingClientRect` that we need to mock somehow).
+
+import BaseComponent from './BaseComponent';
+import { EventHandler, SelectorEngine } from './dom';
+import { debounce, enableToggleAutoloader } from './utils';
+
+const NAME = 'scrollView';
+const DEBOUNCE_DELAY_MS = 50;
+const DEFAULT_DIRECTION = 'vertical';
+const VIEWPORT_SELECTOR = '[data-spirit-element="viewport"]';
+const CONTENT_SELECTOR = '[data-spirit-element="content"]';
+const IS_SCROLLED_AT_START_CLASS_NAME = 'is-scrolled-at-start';
+const IS_SCROLLED_AT_END_CLASS_NAME = 'is-scrolled-at-end';
+const RESIZE_EVENT = 'resize';
+const SCROLL_EVENT = 'scroll';
+
+// Method `getElementsPositionDifference` sometimes returns floating point values that results in inaccurate detection
+// of start/end. It is necessary to accept this inaccuracy and take every value less or equal to 1px as start/end.
+const EDGE_DETECTION_INACCURACY_PX = 1;
+
+export type Direction = 'horizontal' | 'vertical';
+
+interface ICurrentPosition {
+  [key: string]: number;
+  bottom: number;
+  left: number;
+  right: number;
+  top: number;
+}
+
+class ScrollView extends BaseComponent {
+  content: HTMLElement;
+  currentPosition: ICurrentPosition;
+  direction: Direction;
+  isScrolledAtEnd: boolean;
+  isScrolledAtStart: boolean;
+  scrollPositionEnd: string;
+  scrollPositionStart: string;
+  viewport: HTMLElement;
+
+  static get NAME(): string {
+    return NAME;
+  }
+
+  static get DATA_KEY(): string {
+    return `${this.NAME}`;
+  }
+
+  constructor(element: HTMLElement) {
+    super(element);
+
+    this.currentPosition = {
+      bottom: 0,
+      left: 0,
+      right: 0,
+      top: 0,
+    };
+    this.direction = this.element.dataset?.direction
+      ? (String(this.element.dataset.direction) as Direction)
+      : DEFAULT_DIRECTION;
+
+    this.content = SelectorEngine.findOne(CONTENT_SELECTOR, this.element) as HTMLElement;
+    this.viewport = SelectorEngine.findOne(VIEWPORT_SELECTOR, this.element) as HTMLElement;
+
+    this.scrollPositionStart = this.direction === 'horizontal' ? 'left' : 'top';
+    this.scrollPositionEnd = this.direction === 'horizontal' ? 'right' : 'bottom';
+
+    this.isScrolledAtStart = false;
+    this.isScrolledAtEnd = false;
+
+    this.init();
```

Comment:
I am not a big fan of using `init` method in Constructor as the constructor itself is initializing magic method. Also, I prefer using the full name `initialize` instead of an abbreviation. It makes no sense to wrap two methods in this `init` because also the lines above are parts of the initializing.

But we are using this in some places in the code anyway 🤷

Reply:
I just prefer separating this logic from the constructor to some initialization method. That code is then much more readable and much clearer, that you know what is happening where. I do it this way in all components and I have never come across a complaint from your side. Even back then, I suggested @crishpeen in one PR to insert those few lines into the created init() method.

Reply:
In addition, it gives the possibility in some cases to reinitialize the component without having to call the entire instance

Reply:
> I am not a big fan of using init method in Constructor

It's not from my head as we have the `init` method in all our plugins. Should it change systematically and everywhere, it cannot be part of this PR 🙂.

Reply:
I know your objections and this is not a blocker for me.